hil/ext/switches/brocade.py

```diff
@@ -280,6 +286,89 @@ def _disable_native_tag(self, interface):
         url = self._construct_url(interface, suffix='trunk/tag/native-vlan')
         self._make_request('DELETE', url, acceptable_error_codes=(404,))
 
+    def save_running_config(self):
+        url = self.hostname + '/rest/' + 'operational-state/bna-config-cmd'
+        body = '<bna-config-cmd><src>running-config</src><dest>startup-config \
+                </dest></bna-config-cmd>'
+        response = self._make_request('POST', url, data=body)
+
+        # check if it succesfully completed and log it
+        tag = '{urn:brocade.com:mgmt:brocade-ras}session-id'
+        root = etree.fromstring(response.text)
+        session_id = root.find(tag).text
+        if self._check_bna_config_status(session_id):
+            logger.debug('Saved running config to startup config')
+        else:
+            logger.debug('Configuration not saved in reasonable time')
+
+    def get_config(self, config_type):
+        # This is the only method that uses pexpect to connect to the switch
+        # because the REST API does not return the startup config file.
+        # This method is only used by the test suite, the rest of the switch
+        # functionality does not care about this method.
+
+        # this is how we get the running config file using the API
+        # I couldn't find a way to get the startup config file similarly.
+        # url = self.hostname + '/rest/config/' + config_type
+        # response = self._make_request('GET', url)
+
+        from hil.ext.switches._console import login, get_prompts
+
+        # is this safe to assume that there will be http in front of the
```

Comment:
it's a terrible assumption

Reply:
chrome will soon not support address without SSL.

Reply:
oh didnt know that. this switch doesn't have a web interface anyway.

Reply:
well, maybe I used the wrong word. It still supports but marks 'insecure' 🤦‍♂️ 
http://www.eweek.com/security/google-chrome-to-label-sites-using-http-as-insecure-starting-in-july

Reply:
Chrome is irrelevant here anyway, since we're not accessing this through a browser.

Reply:
It would make more sense to just parse the URL properly and extract the host; there are perfectly usable modules in the standard library for this.